### PR TITLE
chore(prql-dotnet): Fix .NET bindings

### PR DESCRIPTION
**Pull request by vanillajonathan**
This fixes the .NET bindings to work with the new compiler library update with the refined API.

The unit test passes, but I ran into some problems marshaling the `CompileResult` struct so the source locations are bogus.

### CODE REVIEW

File: prql-dotnet/PrqlCompiler/PrqlCompiler.cs [hunk 1]
```diff
@@ -38,53 +36,95 @@ public static string Compile(string prqlQuery)
...
+        }
+
+        /// <summary>
+        /// Finds variable references, validates functions calls, determines frames and converts PL to RQ.
+        /// </summary>
+        /// <param name="plJson">A PRQL query.</param>
+        /// <returns>JSON.</returns>
+        /// <exception cref="ArgumentException"><paramref name="plJson"/> is null or empty.</exception>
+        /// <exception cref="FormatException"><paramref name="plJson"/> cannot be compiled.</exception>
+        /// <remarks>https://docs.rs/prql-compiler/latest/prql_compiler/ast</remarks>
```
Comment (max-sixty) [inline prql-dotnet/PrqlCompiler/PrqlCompiler.cs]: +1 on these links, great for reducing docs duplication

File: prql-dotnet/PrqlCompiler/PrqlCompiler.cs [hunk 2]
```diff
@@ -38,53 +36,95 @@ public static string Compile(string prqlQuery)
         /// <exception cref="ArgumentException"><paramref name="prqlQuery"/> is null or empty.</exception>
         /// <exception cref="ArgumentNullException"><paramref name="options"/> is <c>null</c>.</exception>
         /// <exception cref="FormatException"><paramref name="prqlQuery"/> cannot be compiled.</exception>
-        public static string Compile(string prqlQuery, PrqlCompilerOptions options)
+        public static Result Compile(string prqlQuery, PrqlCompilerOptions options)
         {
             if (string.IsNullOrEmpty(prqlQuery))
             {
                 throw new ArgumentException(nameof(prqlQuery));
```
Comment (max-sixty) [inline prql-dotnet/PrqlCompiler/PrqlCompiler.cs]: Is this name supposed to match the name at https://github.com/PRQL/prql/pull/2111/files#diff-628f74a4e1fbc63c5979f552afd14a7043936043a54700da168b3186fa36dc22L39? One has `Null` in...

**Review by max-sixty — APPROVED** [review]
Looking great @vanillajonathan 

(usual caveat about me not knowing C# well...)

File: prql-dotnet/PrqlCompiler/PrqlCompiler.cs (hunk 2 above)
Comment (vanillajonathan) [reply prql-dotnet/PrqlCompiler/PrqlCompiler.cs]: It documents the guard clause which throws if the input parameter `options` is `null`. If any property inside the argument is null that is a different matter.

Comment (max-sixty) [reply prql-dotnet/PrqlCompiler/PrqlCompiler.cs]: Ah yes I see, `ArgumentException` is also on L36. Thanks!

**Comment by vanillajonathan** [comment]
@max-sixty Can you help me move this to /bindings/prql-dotnet/ or /bindings/dotnet/ and merge?

**Comment by max-sixty** [comment]
Great — merging! Thanks @vanillajonathan !

**Pull request merged**